src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -972,23 +1139,26 @@ def resolve_imports_and_packages(
         inline_code,
         all_imports,
         all_packages,
+        udf_level_imports,  # We return this extra dict since all_imports is an empty string when session is None.
         upload_file_stage_location,
         custom_python_runtime_version_allowed,
     )
 
 
 def create_python_udf_or_sp(
     session: "snowflake.snowpark.Session",
+    func: Union[Callable, Tuple[str, str]],
     return_type: DataType,
     input_args: List[UDFColumn],
-    handler: str,
+    handler: Optional[str],
```

Comment:
The use of `Optional` here should also be improved by the same follow up as we discussed above.